lua/wikis/commons/OpponentDisplay.lua

```diff
@@ -358,15 +298,21 @@ OpponentDisplay.propTypes.BlockLiteral = {
 }
 
 ---Displays the name of a literal opponent as a block element.
----@param props {flip: boolean?, name: string, overflow: OverflowModes}
----@return Html
+---@param props {flip: boolean?, name: string, overflow: OverflowModes, additionalClasses: string[]?}
+---@return Widget
 function OpponentDisplay.BlockLiteral(props)
 	DisplayUtil.assertPropTypes(props, OpponentDisplay.propTypes.BlockLiteral)
 
-	return DisplayUtil.applyOverflowStyles(mw.html.create('div'), props.overflow or 'wrap')
+	local block = DisplayUtil.applyOverflowStyles(mw.html.create('div'), props.overflow or 'wrap')
```

Comment:
imo add a function to displayUtil ala
```

function DisplayUtil.getOverflowStyles(mode)
	return {
		overflow = (mode == 'ellipsis' or mode == 'hidden') and 'hidden' or nil,
		['overflow-wrap'] = mode == 'wrap' and 'break-word' or nil,
		['text-overflow'] = mode == 'ellipsis' and 'ellipsis' or nil,
		['white-space'] = (mode == 'ellipsis' or mode == 'hidden') and 'pre' or 'normal',
	}
end
```
that we can use to apply those styles for widgets

that way we can phase out the old function when converting stuff to widgets

additionally (oos) we should check what from DisplayUtil is still used/needed and what we can kick

Reply:
you already did that while i was writing it :)